main.py

```diff
@@ -27,46 +31,105 @@ def setup(self):
+        return PipelineController()
 
     @rest("/v1/pipeline", methods=["POST"])
     @validate_openapi(spec)
     def add_pipeline(self, data):
         """Add pipeline"""
-        return jsonify("Operation successful"), 201
+        log.debug(f"add_pipeline /v1/pipeline content: {data}")
+        try:
+            _id = self.pipeline_controller.insert_pipeline(data)
+        except ValidationError as err:
+            msg = self.error_msg(err.errors())
+            log.debug(f"add_pipeline result {msg} 400")
+            raise BadRequest(msg) from err
+        msg = {"id": _id}
+        log.debug(f"add_pipeline result {msg} 201")
+        return jsonify({"id": _id}), 201
 
     @rest("/v1/pipeline", methods=["GET"])
     def list_pipelines(self):
         """List pipelines"""
-        return jsonify("Operation successful"), 201
+        log.debug("list_pipelines /v1/pipeline")
+        args = request.args.to_dict()
+        status = args.get("status", None)
+        pipelines = self.pipeline_controller.get_pipelines(status)
+        return jsonify(pipelines), 200
 
     @rest("/v1/pipeline/<pipeline_id>", methods=["GET"])
     def get_pipeline(self, pipeline_id):
         """Get pipeline by pipeline_id"""
-        return jsonify("Operation successful"), 201
+        log.debug(f"get_pipeline /v1/pipeline/{pipeline_id}")
+        pipeline = self.pipeline_controller.get_pipeline(pipeline_id)
+        if not pipeline:
+            msg = f"pipeline_id {pipeline_id} not found"
+            log.debug(f"get_pipeline result {msg} 404")
+            raise NotFound(msg)
+        return jsonify(pipeline), 200
 
     @rest("/v1/pipeline/<pipeline_id>", methods=["DELETE"])
     def delete_pipeline(self, pipeline_id):
         """Delete pipeline by pipeline_id"""
-        return jsonify("Operation successful"), 201
+        log.debug(f"delete_pipeline /v1/pipeline/{pipeline_id}")
+        result = self.pipeline_controller.delete_pipeline(pipeline_id)
+        if result == 0:
+            msg = f"pipeline_id {pipeline_id} not found"
+            log.debug(f"delete_pipeline result {msg} 404")
+            raise NotFound(msg)
+        msg = f"Pipeline {pipeline_id} deleted successfully"
+        log.debug(f"delete_pipeline result {msg} 201")
+        return jsonify(msg), 200
 
     @rest("/v1/pipeline/<pipeline_id>/enable", methods=["POST"])
     def enable_pipeline(self, pipeline_id):
         """Enable pipeline"""
-        return jsonify("Operation successful"), 201
+        log.debug(f"enable_pipeline /v1/pipeline/{pipeline_id}/enable")
+        try:
+            self.pipeline_controller.update_status(pipeline_id, "enabled")
+        except NotFound as err:
+            msg = f"Pipeline {pipeline_id} not found"
+            log.debug(f"enable_pipeline result {msg} 404")
+            raise err
+        msg = f"Pipeline {pipeline_id} enabled"
+        log.debug(f"enable_pipeline result {msg} 201")
+        return jsonify(msg), 201
```

Comment:
Could you update this status code as 200 instead? 

On some NApps we ended up with 201 for certain enable/disable operations, but let's only keep 201 when an actual resource is created like you have on `POST /v1/pipeline`. Tiny detail but let's keep it consistent with the NApps that our org maintains. Same comment for the disable operation below.

Reply:
Got it, Changed disable and their respective openapi section.